python/src/py_nvbench.cpp

```diff
@@ -0,0 +1,592 @@
+/*
+ *  Copyright 2025 NVIDIA Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 with the LLVM exception
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.
+ *
+ *  You may obtain a copy of the License at
+ *
+ *      http://llvm.org/foundation/relicensing/LICENSE.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
```

Comment:
Suggested additional comment:

```
// ==================================
// PLEASE SYNC WITH __init__.pyi FILE
// ==================================
// If you modify these bindings, please be sure to update the corresponding type hints in
// ../../cuda/nvbench/__init__.pyi
```

I'm not sure what's the best placement, up here, or near `PYBIND11_MODULE` below.